python-frontend/src/test/java/org/sonar/python/tree/IPythonTreeMakerTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.tree;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Test;
+import org.sonar.plugins.python.api.tree.LineMagic;
+import org.sonar.plugins.python.api.tree.LineMagicStatement;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.parser.RuleTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class IPythonTreeMakerTest extends RuleTest {
+
+  private final IPythonTreeMaker treeMaker = new IPythonTreeMaker();
+
+  @Test
+  public void line_magic() {
+    var parse = parseIPython("print(b)\n" +
+      "a = %alias showPath pwd && ls -a\n", treeMaker::fileInput);
+    assertThat(parse).isNotNull();
+    var lineMagic = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC);
+    assertThat(lineMagic)
+      .isNotNull()
+      .isInstanceOf(LineMagic.class);
+
+    parse = parseIPython("print(b)\n" +
+      "a = %timeit foo(b)\n", treeMaker::fileInput);
```

Comment:
Also here, I think it's important to test what the overall statement translates to (do we have an assignment statement whose RHS is a line magic?).

Reply:
done